src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/HttpClientHandler/HttpClientHandlerIntegration.cs

```diff
@@ -37,21 +33,7 @@ public class HttpClientHandlerIntegration
         public static CallTargetState OnMethodBegin<TTarget, TRequest>(TTarget instance, TRequest requestMessage, CancellationToken cancellationToken)
             where TRequest : IHttpRequestMessage
         {
-            if (IsTracingEnabled(requestMessage.Headers))
-            {
-                Scope scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, requestMessage.Method.Method, requestMessage.RequestUri, IntegrationId, out HttpTags tags);
-                if (scope != null)
-                {
-                    tags.HttpClientHandlerType = instance.GetType().FullName;
-
-                    // add distributed tracing headers to the HTTP request
-                    SpanContextPropagator.Instance.Inject(scope.Span.Context, new HttpHeadersCollection(requestMessage.Headers));
-
-                    return new CallTargetState(scope);
-                }
-            }
-
-            return CallTargetState.GetDefault();
+            return HttpClientHandlerCommon.OnMethodBegin(instance, requestMessage, cancellationToken);
```

Comment:
I think is great to have this common methods!

I was thinking also in move `HttpClientHandler`, `SocketsHttpHandler` and `WinHttpHandler` to a new `Http` parent folder (`AutoInstrumentation\Http\[name]`). Something similar to what I'm doing with the `AdoNet` integrations. (I also want to create a new `Testing` folder for `NUnit` and `XUnit` integrations.

WDYT?

Reply:
Yes, I agree, I think that would make a lot of sense 👍

My concern is whether this would cause breaking issues when there's a mismatch between manual and automatic instrumentation? I don't have a good sense of when those happen yet (as changing an `internal` class caused issues in my testing!)

Reply:
Well all instrumentations within `AutoInstrumentation` folder is CallTarget and that is not GA yet. So we are in a good place to make all changes we want. For CallSite integrations you have to look into `Integrations` folder. I don't think this PR has a CallSite integration isn't ? I would expect it in `Integrations\HttpMessageHandlerIntegration.cs`

Reply:
Ah, oops, didn't notice both were required. In which case yes, that makes sense to update the call targets layout I think. 

I assume I should also add the CallSite integration? Or would we rather only introduce this for CallTarget, given that's what everything's moving to? I'm happy either way, but it probably makes sense to add it to CallSite too I'd guess?

Reply:
Urgh, the integration tests are genuinely failing in CI, but passing when I run locally, and who knows why. Great 😩